### ISSUE TITLE
can't determine machine number format

### ISSUE THREAD

**Issue by shahinamazumdar** [brewsci/homebrew-bio#1450]
CHECKLIST:
- ☑️ Confirmed this is a problem with `brew install`ing one, specific formula and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Homebrew/brew/issues/new.
- ☑️ Ran `brew update` and retried your prior step?
- ☑️ Ran `brew doctor`, fixed all issues and retried your prior step?
- ☑️ Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- ☑️ If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

ISSUE:
**1. What you were trying to do (and why)?**

I was trying to install the dependency libccp4 for coot using homebrew. I need to install the dependencies before I build coot.

**- What happened (include command output)?**

Everything worked fine until the last 15 lines, I think.

```
shine@Shahinas-MacBook-Air coot % brew install libccp4
Updating Homebrew...
==> Downloading https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/libccp4/6.
==> Downloading from https://launchpadlibrarian.net/515632021/libccp4_6.5.1.orig.tar.
######################################################################## 100.0%
==> Installing libccp4 from brewsci/bio
==> ./configure --prefix=/opt/homebrew/Cellar/libccp4/6.5.1 --enable-shared --disable
==> make install
Last 15 lines from /Users/shine/Library/Logs/Homebrew/libccp4/02.make:
In file included from ccp4/csymlib.c:27:
In file included from ccp4/ccp4_parser.h:112:
In file included from ccp4/ccp4_utils.h:29:
In file included from ccp4/ccp4_types.h:22:
ccp4/ccp4_sysdep.h:244:4: error: "Can't determine machine number format"
#  error "Can't determine machine number format"
   ^
1 error generated.
make[1]: *** [ccp4/ccp4_parser.lo] Error 1
make[1]: *** [ccp4/ccp4_general.lo] Error 1
1 error generated.
1 error generated.
make[1]: *** [ccp4/csymlib.lo] Error 1
make[1]: *** [ccp4/cmtzlib.lo] Error 1
make: *** [install] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/brewsci/homebrew-bio/issues

shine@Shahinas-MacBook-Air coot % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /opt/homebrew/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/opt/homebrew/sbin:$PATH"' >> ~/.zshrc

shine@Shahinas-MacBook-Air coot % brew config
HOMEBREW_VERSION: 3.3.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 354718c40a4e7cc4e5d1d4b03fdce8b992e04e8a
Last commit: 31 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 47e6dc9e7fe7284066ee7159cace7201e7e315fb
Core tap last commit: 4 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.30.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.0.1-arm64
CLT: 13.1.0.0.1.1633545042
Xcode: 13.1
Rosetta 2: false
```

I have installed Python 3.10 on my computer. After installing homebrew, I have tapped brewsci/bio, brewsci/science, brewsci/num, brewsci/base. I was basically following instructions from this page:

[http://scottlab.ucsc.edu/xtal/wiki/index.php/Coot_on_10.15_and_11.X](url)

Thanks for your help!

**Comment by jonchang** [comment]
This is likely a problem with the new m1 macs, libccp4 hasn't been updated for them yet. I note that Debian has a patch for this endianness check, if you want to try to apply it and see if it fixes things: https://salsa.debian.org/science-team/libccp4/-/tree/master/debian/patches

**Comment by jongseo-park** [comment]
How about installing CCP4 from the official website ? 
Then coot will be automatically installed.

**Comment by pemsley** [comment]
https://github.com/cctbx/ccp4io/commit/000d8f5a6417fb887c62f631338ddf5f092e55fa

(link from Nick Devenish)

**Comment by wojdyr** [comment]
Here is the corresponding change upstream:
http://fg.oisin.rc-harwell.ac.uk/scm/loggerhead/libccp4/trunk/revision/1885/ccp4/ccp4_sysdep.h

**Comment by YoshitakaMo** [comment]
This issue is fixed by #1493.
